### PR TITLE
feat(constants): Pull Request Status Constants

### PR DESCRIPTION
**Pull request by souravdasslg**
Pull Request statuses are added in `constants/pull-requests.ts`
1. Updated all references
2. Update tests

#5088

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Resolve conflicts

**Review by rarkins — CHANGES_REQUESTED** [review]
Resolve conflicts

**Review by rarkins — CHANGES_REQUESTED** [review]
Rename `PULL_REQUEST_` prefix to `PR_`

File: test/platform/azure/index.spec.ts [hunk 1]
```diff
@@ -212,7 +217,7 @@ describe('platform/azure', () => {
                   pullRequestId: 1,
                   sourceRefName: 'refs/heads/branch-a',
                   title: 'branch a pr',
-                  state: 'open',
+                  state: PR_STATUS_OPEN,
```
Comment (rarkins) [inline test/platform/azure/index.spec.ts]:
```suggestion
                  state: 'open',
```

File: test/platform/azure/index.spec.ts [hunk 2]
```diff
@@ -226,13 +231,13 @@ describe('platform/azure', () => {
             number: 1,
             sourceRefName: 'refs/heads/branch-a',
             title: 'branch a pr',
-            state: 'open',
+            state: PR_STATUS_OPEN,
```
Comment (rarkins) [inline test/platform/azure/index.spec.ts]:
```suggestion
            state: 'open',
```

File: test/platform/azure/index.spec.ts [hunk 3]
```diff
@@ -226,13 +231,13 @@ describe('platform/azure', () => {
...
           } as any)
       );
       const res = await azure.findPr({
         branchName: 'branch-a',
         prTitle: 'branch a pr',
-        state: 'open',
+        state: PR_STATUS_OPEN,
```
Comment (rarkins) [inline test/platform/azure/index.spec.ts]:
```suggestion
        state: 'open',
```

File: test/platform/azure/index.spec.ts [hunk 4]
```diff
@@ -248,7 +253,7 @@ describe('platform/azure', () => {
                   pullRequestId: 1,
                   sourceRefName: 'refs/heads/branch-a',
                   title: 'branch a pr',
-                  state: 'closed',
+                  state: PR_STATUS_CLOSED,
```
Comment (rarkins) [inline test/platform/azure/index.spec.ts]:
```suggestion
                  state: 'closed',
```

File: test/platform/azure/index.spec.ts [hunk 5]
```diff
@@ -262,13 +267,13 @@ describe('platform/azure', () => {
             number: 1,
             sourceRefName: 'refs/heads/branch-a',
             title: 'branch a pr',
-            state: 'closed',
+            state: PR_STATUS_CLOSED,
```
Comment (rarkins) [inline test/platform/azure/index.spec.ts]:
```suggestion
            state: 'closed',
```

File: test/platform/azure/index.spec.ts [hunk 6]
```diff
@@ -262,13 +267,13 @@ describe('platform/azure', () => {
...
           } as any)
       );
       const res = await azure.findPr({
         branchName: 'branch-a',
         prTitle: 'branch a pr',
-        state: '!open',
+        state: PR_STATUS_NOT_OPEN,
```
Comment (rarkins) [inline test/platform/azure/index.spec.ts]:
```suggestion
        state: '!open',
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Revert constants from "fixtures" (i.e. when we're mocking *responses* from platforms).

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs coverage update

**Review by rarkins — CHANGES_REQUESTED** [review]
Rename `PR_STATUS_` to `PR_STATE_`

File: test/platform/github/index.spec.ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import fs from 'fs-extra';
-import { GotApi, GotResponse, RepoConfig } from '../../../lib/platform/common';
+import { GotApi, GotResponse, RepoConfig } from '../../../lib/platform';
```
Comment (rarkins) [inline test/platform/github/index.spec.ts]: Why this change?

**Review by rarkins — CHANGES_REQUESTED** [review]
@souravdasslg could you also add the new `gitea` platform too?

**Comment by souravdasslg** [comment]
> @souravdasslg could you also add the new `gitea` platform too?

Sure. Starting working on that.

File: test/platform/github/index.spec.ts (hunk 7 above)
Comment (souravdasslg) [reply test/platform/github/index.spec.ts]: IDE suggested that improvement, as there is only one file under the platform folder, that is imported automatically.

**Comment by rarkins** [comment]
@souravdasslg can you grant me write access to your fork? I'm trying to push an update to this branch but I'm getting rejected.

**Comment by souravdasslg** [comment]
> @souravdasslg can you grant me write access to your fork? I'm trying to push an update to this branch but I'm getting rejected.

Done! Somehow I forgot to tick the "allow edits from maintainer"
